CHANGELOG.md

```diff
@@ -7,6 +7,188 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+## [16.0.0] - 2023-08-XX
+
+### Breaking changes
+
+-   Now only supporting CDI 4.0
+-   Now supporting FDI 1.18
+-   removed the recipe specific `User` type, now all functions are using the new generic `User` type
+-   The `fetchValue` callback of claims now take a new `recipeUserId` param
+-   Now ignoring protected props in the payload in `createNewSession` and `createNewSessionWithoutRequestResponse`
+-   `createdNewUser` has been renamed to `createdNewRecipeUser` in sign up related APIs and functions
+
+-   EmailPassword:
+    -   removed `getUserById`, `getUserByEmail`
+    -   added `consumePasswordResetToken`
+    -   added an overrideable `createNewRecipeUser` function that is called during sign up and in the “invitation link” flow
+    -   `recipeUserId` is added to the input of `getContent` of the email delivery config
+    -   `email` was added to the input of `createResetPasswordToken` , `sendResetPasswordEmail`, `createResetPasswordLink`
+    -   `updateEmailOrPassword` :
+        -   now takes `recipeUserId` instead of `userId`
+        -   can return the new `EMAIL_CHANGE_NOT_ALLOWED_ERROR` status
+    -   `signIn`:
+        -   returns `recipeUserId`
+    -   `signUp`:
+        -   returns `recipeUserId`
+    -   `signInPOST`:
+        -   can return status `SIGN_IN_NOT_ALLOWED`
+    -   `signUpPOST`:
+        -   can return status `SIGN_UP_NOT_ALLOWED`
+    -   `generatePasswordResetTokenPOST`:
+        -   can now return `PASSWORD_RESET_NOT_ALLOWED`
+    -   `passwordResetPOST`:
+        -   now returns the `user` and the `email` whose password was reset
+        -   can now return `PASSWORD_POLICY_VIOLATED_ERROR`
```

Comment:
this was done before as well. Maybe you meant to use another status code?

Reply:
No, there is a related change in `lib/ts/recipe/emailpassword/types.ts` maybe it was a typing issue before.